### DISCUSSION TITLE
`morph` does not preserve colors

### DISCUSSION THREAD

**Discussion by paya-cz** [quasarframework/quasar#17337] [General - Components / Directives / etc]
I am playing around with `morph`. I noticed that when I morph `q-icon color="negative"`, the color changes to black during the animation and then reverts back to `negative`. Using some basic debugging, it appears that the clone inserted into the DOM does not copy CSS variables properly. The `--q-negative` added to its `style` is `undefined`.

I believe it is because of how the CSS property values are retrieved:

https://github.com/quasarframework/quasar/blob/19157891a1271c96ee67c286cf0c93874e94d3a0/ui/src/utils/morph/morph.js#L57

This line returns `undefined` for the `--q-negative` property in my browser (Firefox). But if I use [getPropertyValue](https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/getPropertyValue) then the correct value is retrieved (`#1976d2`). I observe similar behavior in Safari.

Alternatively, the CSS variables should not be added to the clone `style` in the first place, to prevent the `:root` variables from being overridden by `undefined`.

I made a little demo here: https://stackblitz.com/edit/quasarframework-9vvypu?file=src%2Fpages%2FIndexPage.vue

IndexPage.vue:

```
<template>
  <q-page class="column flex-center">
    <q-knob
      v-model="count"
      :min="min"
      :max="max"
      size="80px"
      show-value
      :thickness="0.13"
      color="primary"
      track-color="dark"
    >
      <q-avatar size="75px">
        <img alt="Quasar logo" src="~assets/quasar-logo-inner.svg" />
      </q-avatar>
    </q-knob>

    <img
      alt="Quasar logo"
      src="~assets/quasar-logo-vertical.svg"
      style="width: 200px; height: 140px"
    />

    <div class="row no-wrap q-gutter-x-lg items-center relative-position">
      <q-btn
        color="primary"
        no-wrap
        label="Morph element"
        @click="morphContent1"
      ></q-btn>

      <div ref="firstMorphRef">
        <div v-bind="props1">
          <q-icon name="close" color="positive" size="3rem" />
          {{ toggle1 ? 'Small' : 'Large' }}
        </div>
      </div>
    </div>
  </q-page>
</template>

<script setup>
import { ref, computed } from 'vue';
import { morph } from 'quasar';

const toggle1 = ref(false);
const firstMorphRef = ref(null);
let cancel1;

const props1 = computed(() => {
  return toggle1.value === true
    ? {
        class: 'q-ml-sm q-pa-md bg-orange text-white rounded-borders',
        style: 'font-size: 24px',
      }
    : {
        class: 'q-ml-xl q-px-xl q-py-lg bg-blue text-white',
        style: 'border-radius: 25% 0/50% 0; font-size: 36px',
      };
});

function morphContent1() {
  const onToggle = () => {
    toggle1.value = toggle1.value !== true;
  };

  if (cancel1 === void 0 || cancel1() === false) {
    cancel1 = morph({
      from: firstMorphRef.value,
      onToggle,
      duration: 500,
      tween: true,
      onEnd: (end) => {
        end === 'from' && onToggle();
      },
    });
  }
}
</script>
```

I noticed yet another problem with `morph`. If I have a `span` use `text-red-9` class to work around the broken `text-negative` class, the morph adds `-webkit-text-fill-color` property on the target element which overrides the `color` set by the `text-red-9`, making the text black despite `text-red-9`.

**Comment by ivanjaros** [comment]
Looks more like a bug suited for issue ticket rather than discussion.
